src/main/java/org/mockito/internal/util/Platform.java

```diff
@@ -0,0 +1,48 @@
+package org.mockito.internal.util;
+
+public abstract class Platform {
+
+    public static final String JAVA_VERSION = System.getProperty("java.specification.version");
+    public static final String JVM_VERSION = System.getProperty("java.runtime.version");
+    public static final String JVM_VENDOR = System.getProperty("java.vm.vendor");
+    public static final String JVM_VENDOR_VERSION = System.getProperty("java.vm.version");
+    public static final String JVM_NAME = System.getProperty("java.vm.name");
+    public static final String JVM_INFO = System.getProperty("java.vm.info");
+
+    private Platform() {}
+
+    public static String describe() {
+        return String.format("Java               : %s\n" +
+                             "JVM vendor name    : %s\n" +
+                             "JVM vendor version : %s\n" +
+                             "JVM name           : %s\n" +
+                             "JVM version        : %s\n" +
+                             "JVM info           : %s\n" +
+                             "OS name            : %s\n" +
+                             "OS version         : %s\n"
+                ,
+                             JAVA_VERSION,
+                             JVM_VENDOR,
+                             JVM_VENDOR_VERSION,
+                             JVM_NAME,
+                             JVM_VERSION,
+                             JVM_INFO,
+                             System.getProperty("os.name"),
+                             System.getProperty("os.version"));
+    }
+
+    public static boolean isJava8BelowUpdate45() {
+        if(!JAVA_VERSION.equals("1.8")) {
+            return false;
+        }
+
+        int beginIndex = JVM_VERSION.indexOf('_');
+        int endIndex = JVM_VERSION.indexOf('b');
+        if(beginIndex == -1 || endIndex == -1) {
+            return false;
+        }
+
+        String updateVersion = JVM_VERSION.substring(beginIndex + 1, endIndex - 1);
+        return 45 > Integer.valueOf(updateVersion);
```

Comment:
Let's make sure this is always safe, regardless of the content of the system properties. Those properties are mutable AFAIK, and/or might change/be different with future versions of java or perhaps even jvm/jdk vendor.

Otherwise, love this PR :)